### ISSUE TITLE
Bug: [no-meaningless-void-operator] Ignores non-call expressions

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#12214]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=6.0.2&fileType=.ts&code=CYUwxgNghgTiAEYD2A7AzgF3gIyQDwC54BveANyggFcQjMYBLFAc3gF8BuAKC7KQeA583PgKF4AdBWogR-QbknSaEjEgCqAB00gYAYShoQACgCU3IA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oFtEBDJswDmSZMloA3DpQAmtDqWh98HaOiiJo0VZHBgAviH1A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
declare const box: { value: string };

void box;
void box.value;
void box.value.toUpperCase();
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-meaningless-void-operator": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

Each of those three expressions should be reported as a meaningless `void` expression.

### Actual Result

None are reported.

### Additional Info

Note that `void` before a call expression _does_ get reported on:

```ts
declare const getValue(): string;
void getValue();
```

I've seen AIs write code using `void`s to get around no-unused-expressions. I hate this.

💩 😡
